plugins/search/src/components/SearchResult/SearchResult.tsx

```diff
@@ -65,9 +66,14 @@ type Filters = {
 // TODO: move out column to make the search result component more generic
 const columns: TableColumn[] = [
   {
-    title: 'Component Id',
+    title: 'Name',
     field: 'name',
     highlight: true,
+    render: (entity: Partial<Result>) => (
+      <Link to={`/catalog/${entity.namespace}/${entity.kind}/${entity.name}`}>
```

Comment:
oh actually, just realized [namespace can be optional](https://backstage.io/docs/architecture-decisions/adrs-adr002#namespace) now, so need to change that here

Reply:
I don't think they're supposed to be optional in the URL, and should be `default` if the entity doesn't have a namespace. Not sure that's how it's implemented in the catalog though. https://backstage.io/docs/architecture-decisions/adrs-adr009#entity-references-in-urls

Reply:
Would be cool if entities had methods like `getBaseUrl()` or something so we didn't have to fiddle with / know these kinds of things.

Reply:
^^ really good idea

Reply:
I recently learnt about `getBaseUrl` in backend plugins using [`discovery`](https://github.com/backstage/backstage/tree/master/packages/backend-common/src/discovery) APIs. I am guessing there is [`DiscoveryApi`](https://github.com/backstage/backstage/blob/master/packages/core-api/src/apis/implementations/DiscoveryApi/UrlPatternDiscovery.ts) for frontend too.

Reply:
Hmmm interesting, so I think here we want to get a url for each entity and not actually a plugin, the DiscoveryApi seems to be responsible for constructing URLs for each plugin. 🤔 Or I could be wrong here

Reply:
Definitely not ON entities - but yes. We are right now settling how routing and page composition is to work in the final non-alpha product, and this sounds like one of those places where route refs come into play.

Scarce old info here: https://backstage.io/docs/reference/createPlugin-router

The route refs will probably remain (still working on the details) and you'd essentially be using a facility to resolve the target URL using them. There would be something like this one https://github.com/backstage/backstage/blob/master/plugins/catalog/src/plugin.ts#L25 but for the entity landing page.

Reply:
To get the three parts out of an entity including the default, you can use this https://github.com/backstage/backstage/blob/master/packages/catalog-model/src/entity/ref.ts#L28

Reply:
Hmmm @freben I don't know if it makes sense here to add catalog model as a dependency only because I want the default fallback 🤔 I already have access to name, namespace and kind, [see changes.](https://github.com/backstage/backstage/pull/3349/files#diff-7dd9aeca72a846e6cd397452886389314a602a138c39d8d545c1bb49c45e25e9R45)

Reply:
If it makes it easier to stomach, that package is very special - it is ultra light weight on its own depencencies and is special made to be imported everywhere by both frontend and backend packages because it holds the "master" types to deal with entities for everybody. So if you use the Entity type, for example, you are actually already using it from that one.

Reply:
Also that string `'default'` exists as an exported const from the same package for the same reasons :)

Reply:
Hey @freben thanks for response, I don't use the Entity type yet but I probably should, so that gave me at least one more reason to use it. 🎉